### ISSUE TITLE
Entity Framework Migration and Persisted Volumes

### ISSUE THREAD

**Issue by dcaponi** [dotnet/docs-aspire#103]
**Unclear How to Set up and Migrate pSQL DB w/ EntityFramework DB Contexts**

Hey team, I'm trying to follow along with the docs but instead of a redis cache and 2 apps (UI and API) I'm just doing API & Postgres using Npgsql.

~How does the connection string work here? I'd think it was all handled and I wouldn't have to configure a connection string for local dev (the redis setup makes no mention of a connection string). Do I need to add that to the `AppHost` or API's `appsettings.json`?~

It looks like the port changes for the database container every time I restart the Aspire project. Any way to make that port static so I don't have to remember to change it each time?

~Finally, it would be neat to get more details about how to manage migrations for entity framework. I assume I'd just run `dotnet ef database update` from the API project once I stood up the project but theres that pesky connection string issue.~

I spin up the project with dotnet run --project AspireSample.AppHost and in the API project I add the connection string to appsettings.json like so `"db": "Host=localhost;Port=51586;Database=postgres;User ID=postgres;Password=postgres"` then in another terminal from the API project I run `dotnet ef database update` to kick off the migration.

Otherwise, really neat project and good docs 🙂

**.NET Core 8**

- [x] .NET Core
- [ ] .NET Framework
- [ ] .NET Standard

<details>
<summary><strong>
.NET SDK:
 Version:           8.0.100
 Commit:            57efcf1350
 Workload version:  8.0.100-manifests.71b9f198

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  14.1
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.100/

.NET workloads installed:
 Workload version: 8.0.100-manifests.71b9f198
 [aspire]
   Installation Source: SDK 8.0.100
   Manifest Version:    8.0.0-preview.1.23557.2/8.0.100
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.aspire/8.0.0-preview.1.23557.2/WorkloadManifest.json
   Install Type:        FileBased

Host:
  Version:      8.0.0
  Architecture: arm64
  Commit:       5535e31a71

.NET SDKs installed:
  8.0.100 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.NETCore.App 8.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]

</strong></summary>

```console
<replace>
```
</details>

**Comment by dcaponi** [comment]
Update: So I spin up the project with `dotnet run --project AspireSample.AppHost` and in the api project where I define my DbContext and models I add the connection string to `appsettings.json` like so `"db": "Host=localhost;Port=51586;Database=postgres;User ID=postgres;Password=postgres"` 

I had to fish the port out of the aspire dashboard to paste on my connection string in order to get the migration to work. I ran the migration from my api project with `dotnet ef database update`

So I don't know if I'm making a rookie .NET mistake here so feel free to flame and/or correct me here, but one thing I can see getting really annoying is having to remember to update the port in the connection string every time I restart the aspire project since it seems the exposed port in the docker container can change. I can see an argument for not pinning it to 5432 but I'd like to make it static to something so future me doesn't have to hunt it down every time.
